plugins/autoscroll.koplugin/main.lua

```diff
@@ -0,0 +1,132 @@
+local Device = require("device")
+local Event = require("ui/event")
+local PluginShare = require("pluginshare")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local AutoScroll = WidgetContainer:new{
+    name = 'autoscroll',
+    is_doc_only = true,
+    auto_scroll_sec = G_reader_settings:readSetting("auto_scroll_timeout_seconds") or 0,
+    settings_id = 0,
+    last_action_sec = os.time(),
+}
+
+function AutoScroll:_enabled()
+    return self.auto_scroll_sec > 0
```

Comment:
This makes more sense to me as a completely separate enabled setting, but for starters I just wanted to get the base functionality working, and deviating from AutoSuspend would've taken more time.